src/composables/use-delete-account-alert.ts

```diff
@@ -0,0 +1,37 @@
+import Swal from 'sweetalert2';
+import { i18n } from '@/plugins/i18n';
+import { auth } from '@/store/auth';
+import { UsersApi } from '@/api';
+import { useApi } from '@/composables/apis';
+
+/**
+ * Esta función se encarga de preguntar al usuario si quiere borrar su cuenta con un mensaje
+ */
+export async function deleteAccountMessage() : Promise<void> {
+  const { t } = i18n;
+
+  await Swal.fire({
+    title: t('deleteAccountMessageTitle'),
+    text: t('deleteAccountMessageText'),
+    icon: 'warning',
+    showCancelButton: true,
+    confirmButtonColor: '#910e38',
+    cancelButtonColor: '#0e4781',
+    confirmButtonText: t('deleteAccountMessageConfirm'),
+    cancelButtonText: t('deleteAccountMessageCancel')
+  }).then(async (res) => {
+    if (res.isConfirmed) {
+      await Swal.fire({
+        title: t('deleteAccountMessageConfirmedTitle'),
+        text: t('deleteAccountMessageConfirmedText'),
+        icon: 'success'
+      });
+      await useApi(UsersApi, 'usersMeDeleteDestroy')();
+      auth.logout();
+
+      return 'Account deleted';
+    } else {
+      return 'Account deletion cancelled';
+    }
+  });
```

Comment:
```suggestion
  }).then(async (res) => {
    if (res.isConfirmed) {
      await Swal.fire({
        title: t('deleteAccountMessageConfirmedTitle'),
        text: t('deleteAccountMessageConfirmedText'),
        icon: 'success'
      });
      await useApi(UsersApi, 'usersMeDeleteDestroy')();
      auth.logout();

      return 'Account deleted';
    } else {
      return 'Account deletion cancelled';
    }
  });
```

El then no lo uses, es sintaxis antigua de Javascript.

Lo que has hecho es lo mismo que:

```
const res = await Swal.fire(....

if (res.confirmed) {
....resto del código que está dentro del then
````

Aparte, por qué devuelves "Account deletion" y "Account deletion canceled", no se usa el valor retornado por esta función.

Reply:
Con el then me pedía que hubiera return sí o sí. Ya lo he cambiado.